### PR TITLE
EVP: Enforce that EVP_PKEY_set_alias_type() only works with legacy keys

### PR DESCRIPTION
**Pull request by levitte**
This also deprecates the function, as it is not necessary any more,
and should fall out of use.

### CODE REVIEW

**Comment by levitte** [comment]
This is the opposite of #12675

**Review by romen — DISMISSED** [review]
Travis failure is relevant, but it is probably just a matter of allowing deprecated functions in the failing file.

If that is the only change required and CI passes otherwise, this LGTM.

**Comment by levitte** [comment]
The Travis issue will go away as soon as #12536 is merged

**Comment by mattcaswell** [comment]
Assuming this should have the "approval: done" label, so adding it now.

**Comment by levitte** [comment]
I needed to make a small change in `test/evp_extra_test.c`.  Do note that we already test `EVP_PKEY_set_alias_type()` quite thoroughly in `test/ecdsatest.c`.

File: CHANGES.md
Comment (slontis) [inline CHANGES.md:36]: This is fairly difficult to read.
What about simplifying it to..
``` 
* Deprecated EVP_PKEY_set_alias_type().  This function was previously needed as a workaround for SM2. With OpenSSL 3.0, this key type is internally recognized
so the workaround is no longer needed.
Functionality is still retain as it is, but will only work with EVP_PKEYs with a legacy internal key.

```

File: crypto/evp/p_lib.c
Comment (slontis) [inline crypto/evp/p_lib.c:667]: Isn't this likely to  break existing applications that call this function?

File: CHANGES.md
Comment (levitte) [reply CHANGES.md:36]: Sure

File: crypto/evp/p_lib.c
Comment (levitte) [reply crypto/evp/p_lib.c:667]: Uhm, depends on the internal key. What would you have this do if it's called with a provider-native key?

File: CHANGES.md
Comment (levitte) [reply CHANGES.md:36]: Done

File: crypto/evp/p_lib.c
Comment (levitte) [reply crypto/evp/p_lib.c:667]: I think I caught your thinking, and added an extra check to see if the key type is what the user asks for, and return 1 in that case.

**Comment by slontis** [comment]
Travis failure is just a timeout..

**Review by slontis — APPROVED** [review]
LGTM

**Comment by levitte** [comment]
Merged

14711fffbf2726f94d6a05d8f9cbae7fe7a4bc09 EVP: Enforce that EVP_PKEY_set_alias_type() only works with legacy keys
8a288609b12488b5e2dbca28ea54c3dd2c0de15b TEST: Remove use of EVP_PKEY_set_alias_type() in test/evp_extra_test.c

**Pull request closed without merging**
